### DISCUSSION TITLE
using python-benedict with jello

### DISCUSSION THREAD

**Discussion by g-v-egidy** [kellyjonbrazil/jello#62] [Ideas]
Today I stumpled upon the python-benedict module: https://github.com/fabiocaccamo/python-benedict
and I immediately thought of using it with jello.

It allows to use the same keys as attributes (dotted notation) access like jello. But on top of that it contains several utility functions that make restructuring, searching and manipulating dicts easier:
https://github.com/fabiocaccamo/python-benedict#utility-methods

So this seems to me like a very good match to extend jello.

Of course one could always manually create a benedict object out of _. This would work without changes to jello.

But it might also be possible to automatically create _ as a benedict object when a special commandline switch is given for example.

What do you think about this?

**Comment by kellyjonbrazil** [comment]
I like the idea but also need to balance with not adding too many features and dependencies into `jello`.

I have added the ability to ingest raw string data via `-R` in https://github.com/kellyjonbrazil/jello/issues/64. This might be a good way to allow the use of other libraries without adding the features/dependencies into `jello`.

Here is an example of loading YAML data into `jello`:
```
% jello -R 'import yaml;_ = yaml.safe_load(_)' -f values.yaml 
{
  "var1": "value1",
  "var2": "value2",
  "text": "Here a text\nthat i like to write like this on multiple line.\nIt will be an HTML text so i’ll add <br> for line return.\nand here i finish"
}
```
What do you think?

**Comment by g-v-egidy** [comment]
Would you consider a patch for a `-b` cli option for jello that uses the standard dict input (as it is done now), but creates a benedict object for `_`?

The -b option would then only import the benedict python module when it is actually used, so there would be no hard dependency on benedict.

Having this would make the additional dict-methods from benedict available to jello without requiring the user to add boilerplate code that converts `_` to benedict first.
